File: object-oriented-programming/blackjack-game/python/deck_class.py
# Isaac Computer Science
# Usage licensed under the Open Government Licence v3.0

from card_class import PlayingCard

SUITS = ["Hearts", "Spades","Clubs", "Diamonds"]
RANKS = ["Ace", "2", "3", "4", "5", "6", "7", "8", "9", "10", "Jack", "Queen", "King"]
VALUES = [11, 2, 3, 4, 5, 6, 7, 8, 9, 10, 10, 10, 10]

import random

class Deck:
    """Defines a standard 52-card deck of cards"""

    def __init__(self):
        """Constructor"""
        self.reset()
        
    
    def shuffle(self):
        """Shuffles the cards"""
        random.shuffle(self.__cards)
    

    def deal(self):
        """Removes and returns the first card in the deck"""
        return self.__cards.pop()

    def reset(self):
        """Resets the deck"""
        self.__cards = []
        for i in range(4):
            for j in range(13):
                new_card = PlayingCard(SUITS[i], RANKS[j], VALUES[j])
                self.__cards.append(new_card)
        self.shuffle()


# This code will run if this file is executed directly
# (i.e. not called by another program)

if __name__ == '__main__':       
    my_deck = Deck()  # Test instantiation

File: object-oriented-programming/fundamentals/c-sharp/deck_class.cs
/*
Isaac Computer Science
Usage licensed under the Open Government Licence v3.0

Note: This file is designed to be copied out and compiled on your machine.
In order for it to compile properly you need to ensure that the project name is the same as the "namespace" in this file.
To run this file you need to:
1. Copy the contents
2. Paste them into the C# IDE of your choice (Visual Studio, for example)
3. Change the namespace to match your project (if necessary)
4. Compile the program
5. Run the program
*/

using System;
using System.IO;
using card_class;

namespace IsaacCodeSamples
{
    class Deck
    {
        
        private string[] cards = new string[52]; // Declare an array of 52 elements
        private string[] suits = ["Clubs", "Diamonds", "Hearts", "Spades"]
        private string[] ranks = ["Ace", "2", "3", "4", "5", "6", "7", "8", "9", "10", "Jack", "Queen", "King"]
        private string[] values = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 10, 10, 10]

        // Constructor method
        public static void Deck() {
            for (int i = 0; i = 4; i++) {
                for (int j = 0; j = 13; j++) {
                new card = Card(suits[i], ranks[j], values[j])
                this.cards.append(card);
                }
             }
             
        }

    }
    
    
    class Testing
    {
        // The Main method is the entry point for all C# programs
        public static void Main()
        {
            myDeck = new Deck();  // Instantiate a new deck object
        }
    }
    
    
}
